utils/perf_feature_stats.py

```diff
@@ -0,0 +1,60 @@
+"""Place to store any perf functions for feature benchmark tests.
+"""
+from utils.log import logger
+from utils.path import log_path
+from utils.conf import cfme_data
+import numpy
+import os
+
+
+def get_benchmark_providers():
+    """Gets all providers from cfme_data with tag 'benchmark'."""
+    providers = []
+    for provider in cfme_data['management_systems']:
+        if 'benchmark' in cfme_data['management_systems'][provider]['tags']:
+            providers.append(provider)
+    return providers
+
+
+def log_stats(timings, feature, test, provider):
+    """Dumps raw timing values and wites/appends statistics to feature-statistics.csv"""
+    csv_prefix = '{}-{}-{}'.format(feature, test, provider)
+
+    csv_path = log_path.join('csv_output')
+    if not os.path.exists(str(csv_path)):
+        os.mkdir(str(csv_path))
+
+    # Dump Raw Values:
+    csv_file_path = csv_path.join(csv_prefix + '-timings.csv')
+    with open(str(csv_file_path), 'w') as csv_file:
+        csv_file.write(csv_prefix + '\n')
+        for timing in timings:
+            csv_file.write(str(timing) + '\n')
+
+    numpy_arr = numpy.array(timings)
+    minimum = numpy.amin(numpy_arr)
+    average = numpy.average(numpy_arr)
+    median = numpy.median(numpy_arr)
+    maximum = numpy.amax(numpy_arr)
+    stddev = numpy.std(numpy_arr)
+    percentile90 = numpy.percentile(numpy_arr, 90)
+    percentile99 = numpy.percentile(numpy_arr, 99)
+
+    # Write/Append to features benchmark csv
+    csv_file_path = csv_path.join('features-statistics.csv')
+    if csv_file_path.isfile():
+        logger.info('Appending to: features-statistics.csv')
+        outputfile = csv_file_path.open('a', ensure=True)
+    else:
+        logger.info('Writing to: features-statistics.csv')
+        outputfile = csv_file_path.open('w', ensure=True)
+        outputfile.write('feature,test,provider,runs,minimum,average,median,maximum,stddev,90th'
+            ',99th\n')
+
+    outputfile.write('{},{},{},{},{},{},{},{},{},{},{}\n'.format(feature, test, provider,
+        len(timings), minimum, average, median, maximum, stddev, percentile90, percentile99))
```

Comment:
But that still leaves door open for escape-related things to sneak in. I know it is probably not happening here but we have a standard library here so why not use it :)